### ISSUE TITLE
Generated text blocks for view sources should escape \

### ISSUE THREAD

**Issue by nstrong-scw** [jOOQ/jOOQ#14141]
### Expected behavior

Expected behavior: the code generator should generate valid Java code.

### Actual behavior

The regex is inserted into the code unmodified, resulting in the unescaped backslash triggering a compile error during the build phase.

This is a regression in 3.17.x; the same database schema converts correctly with 3.16.10.

### Steps to reproduce the problem

1. Create a postgres database with a table that looks something like this:

    ```sql
    CREATE TABLE example_table (
        name text,
        maybe_number text
    );
    
    CREATE VIEW example_view AS
      SELECT name, description FROM example_table
      WHERE name = 'foo' AND maybe_number ~ '^\d+$'::text;
    ```
2. Run jooq against it
3. Inspect the generated code. You'll get something resembling this:
```java
    private ExampleView(Name alias, Table<ExampleViewRecord> aliased, Field<?>[] parameters) {
        super(alias, null, aliased, parameters, DSL.comment(""), TableOptions.view("""
        CREATE VIEW example_view AS
          SELECT name, description FROM example_table
          WHERE name = 'foo' AND maybe_number ~ '^\d+$'::text;
                                            --  ^^^^^^^ should be '^\\d+$'
        """));
    }
```
[jooq-config.xml.txt](https://github.com/jOOQ/jOOQ/files/9874723/jooq-config.xml.txt)

### jOOQ Version

3.17.4

### Database product and version

TestContainer running Postgres 12

### Java Version

17

### OS Version

Mac OS 13.0 Ventura

### JDBC driver name and version (include name if unofficial driver)

org.testcontainers.jdbc.ContainerDatabaseDrive

**Comment by lukaseder** [comment]
Thanks a lot for your report. I can confirm, this isn't correct, currently, when generating text blocks. You can turn off the generation of text blocks as a workaround, using:

```xml
<generate>
  <textBlocks>false</textBlocks>
</generate>
```

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.18.0 and 3.17.5 (https://github.com/jOOQ/jOOQ/issues/14142). Thanks again for your report!
